### PR TITLE
accepting ph asset uri as background video

### PR DESCRIPTION
**Pull request by Gokaysim**
# Overview

# Test Plan

### CODE REVIEW

**Review by Gokaysim — COMMENTED** [review]
minor bug fix

File: ios/InstagramStories.m [hunk 1]
```diff
@@ -159,8 +160,60 @@ - (void)shareSingle:(NSDictionary *)options
...
             } else {
-                NSData *backgroundVideo = [[NSFileManager defaultManager] contentsAtPath: URL];
-                [self backgroundVideo: backgroundVideo];
+                if([URL hasPrefix:@"ph://"]){
+                    NSURL *imageURL = [NSURL URLWithString:URL];
+                    PHFetchResult *results;
+                    NSString *assetID = @"";
+                    if (!imageURL) {
+                    RCTLogError(@"Cannot load a photo library asset with no URL");
+                    return;
```
Comment (MateusAndrade) [inline ios/InstagramStories.m:169]:
```suggestion
                      RCTLogError(@"Cannot load a photo library asset with no URL");
                      return;
```

File: ios/InstagramStories.m [hunk 2]
```diff
@@ -159,8 +160,60 @@ - (void)shareSingle:(NSDictionary *)options
...
+                    } else if ([imageURL.scheme caseInsensitiveCompare:@"assets-library"] == NSOrderedSame) {
+                    assetID = [imageURL absoluteString];
+                    results = [PHAsset fetchAssetsWithALAssetURLs:@[imageURL] options:nil];
```
Comment (MateusAndrade) [inline ios/InstagramStories.m:172]:
```suggestion
                      assetID = [imageURL absoluteString];
                      results = [PHAsset fetchAssetsWithALAssetURLs:@[imageURL] options:nil];
```

File: ios/InstagramStories.m [hunk 3]
```diff
@@ -159,8 +160,60 @@ - (void)shareSingle:(NSDictionary *)options
...
+                    } else {
+                    assetID = [imageURL.absoluteString substringFromIndex:@"ph://".length];
+                    results = [PHAsset fetchAssetsWithLocalIdentifiers:@[assetID] options:nil];
```
Comment (MateusAndrade) [inline ios/InstagramStories.m:175]:
```suggestion
                      assetID = [imageURL.absoluteString substringFromIndex:@"ph://".length];
                      results = [PHAsset fetchAssetsWithLocalIdentifiers:@[assetID] options:nil];
```

File: ios/InstagramStories.m [hunk 4]
```diff
@@ -159,8 +160,60 @@ - (void)shareSingle:(NSDictionary *)options
...
+                    }
+                    if (results.count == 0) {
+                      RCTLogError(@"%@", [NSString stringWithFormat:@"Failed to fetch PHAsset with local identifier %@ with no error message.", assetID]);
+                    return;
```
Comment (MateusAndrade) [inline ios/InstagramStories.m:179]:
```suggestion
                      return;
```

File: ios/InstagramStories.m [hunk 5]
```diff
@@ -159,8 +160,60 @@ - (void)shareSingle:(NSDictionary *)options
...
+                        {
+                            [self backgroundVideo: data];
+                        }
+                    }];
+                    [dataTask resume];
+                }
+                else{
+                     NSData *backgroundVideo = [[NSFileManager defaultManager] contentsAtPath: URL];
+                    [self backgroundVideo: backgroundVideo];
+                }
```
Comment (MateusAndrade) [inline ios/InstagramStories.m:216]:
```suggestion
                } else {
                     NSData *backgroundVideo = [[NSFileManager defaultManager] contentsAtPath: URL];
                     [self backgroundVideo: backgroundVideo];
                }
```

File: ios/InstagramStories.m [hunk 6]
```diff
@@ -159,8 +160,60 @@ - (void)shareSingle:(NSDictionary *)options
...
+
+                    [[PHImageManager defaultManager] requestAVAssetForVideo:(asset) options:(options) resultHandler:^(AVAsset * _Nullable Asset, AVAudioMix * _Nullable audioMix, NSDictionary * _Nullable info) {
+                    AVURLAsset* myAsset = (AVURLAsset*)Asset;
+
+                    NSData * backgroundVideo = [NSData dataWithContentsOfFile:myAsset.URL.relativePath];
+                    [self backgroundVideo: backgroundVideo];
+
+                    }];
+                }
+                else if([URL hasPrefix:@"http"]){
```
Comment (MateusAndrade) [inline ios/InstagramStories.m:197]:
```suggestion
                }  else if([URL hasPrefix:@"http"]) {
```

**Review by MateusAndrade — CHANGES_REQUESTED** [review]

**Comment by MateusAndrade** [comment]
Hey @Gokaysim thanks for your help, would mind filling in what these changes address or what is the purpose? Also, how we would end using this? 

I saw that we are using the ` Photo`  SDK from iOS, we use this to access the local file and pass it directly to `Instagram`?

**Comment by Gokaysim** [comment]
if the backgroundvideo URI is given as PHAsset it fetches the file with it. If the file is given as http url it downloads it in a different thread.

**Comment by vforvasile** [comment]
@MateusAndrade  how come this isn't merged?

**Pull request closed without merging**